gulpfile.js/lib/webpack-multi-config.js

```diff
@@ -32,6 +32,18 @@ module.exports = function(env) {
     }
   }
 
+  if(env === 'development') {
+    webpackConfig.devtool = 'inline-source-map'
+
+    // Create new entries object with webpack-hot-middleware added
+    for (var key in config.tasks.js.entries) {
+      var entry = config.tasks.js.entries[key]
+      config.tasks.js.entries[key] = ['webpack-hot-middleware/client?&reload=true'].concat(entry)
```

Comment:
The `?&reload=true` is the important bit I was missing on other projects. This forces a full page refresh if hot-module replacement can't be used. This will be the case for most people out of the box.